### ISSUE TITLE
Change equalizer frequencies to be ISO 266 compliant

### ISSUE THREAD

**Issue by santigl** [clementine-player/Clementine#5286]
This is a proposal to modify the equailzer to follow the [ISO 266](www.zytrax.com/tech/audio/equalization.html#iso-frequencies) standard.

According to that standard, the center frequencies are doubled at each step: 30, 60, 130, 250, 500, 1000, 2000, 4000, 8000, 16000 [Hz]. (Actually, those are the human-friendly labels, the actual frequencies are, approximately, 31.62, 63.10, 125.89, 251.19, 501.19, 1000.0, 1995.3, 3981.1, 7943.3, and 15848.3 [Hz].)

The bandwidth of each band of the equalizer is variable, and it is calculated by a simple formula.

As discussed in issue #695, it seems that current equalizer presets are calculated from that standard, so the only issue with the migration would be user-generated presets.

The code with these changes is [here](https://github.com/santigl/Clementine/commit/3d3f2243cd66c110430f61e6f001850d54d45844).

**Comment by Ferroin** [comment]
+1 on changing this as:
1. There appears to be more software that uses the ISO 266 EQ bands.
2. The current curves match (within reason) those used by other software that uses the ISO bands, but sound different due to the band differences.

**Comment by rfgiusti** [comment]
> [...] so the only issue with the migration would be user-generated presets.

Is there a way to migrate without breaking user presets?

If not, is there a clear advantage for changing that that would justify breaking user presets?

**Comment by Ferroin** [comment]
In theory, we could provide a way to interpolate existing user defined presets.  This would not give an identical sound for multiple reasons unless the curve fits some very specific patterns (for example, if it could be directly described by a linear function, it should sound almost identical).

As far as advantages, see my earlier comment.  The biggest one is consistency with a wider variety of software and hardware.

**Comment by santigl** [comment]
And also take into consideration that until #5285 is applied (fixing #695) all presets are broken. Users will still have to manually correct the last band values from their personalized presets.

**Comment by narunlifescience** [comment]
I think it is better to break the preset only once by migrating to ISO 266 & combining #5285.

**Comment by Ferroin** [comment]
I agree, although this got split out from #5277 and the discussion in #695 because there seemed to be enough disagreement about whether or not to switch to these bands.to warrant a separate issue.

**Comment by narunlifescience** [comment]
well.. if ISO 266 makes it to clementine then it should go along with #5285.

**Comment by Ferroin** [comment]
@hatstand @ArnaudBienner, do either of you have any preference regarding whether this should be merged together with #5285 or not (not necessarily whether or not we should switch the bands, but whether or not we want to potentially break presets twice)?

**Comment by santigl** [comment]
With #5285 merged, issue #695 is now fixed.

We are now in the path to breaking user-defined presets (in their last bands). Should we take the opportunity and make the change to ISO frequencies as well?

**Comment by Ferroin** [comment]
I would vote yes, but that's just me.

Ideally, I'd really like to see opinions on this from @hatstand and/or @ArnaudBienner.

**Comment by rfgiusti** [comment]
To the best of my knowledge, the break caused by #5285 is minimal, while
changing to ISO frequencies would probably break user-defined presets
completely.

So I would like to argue not to make this change unless there is real
benefit in adopting ISO frequencies. Don't just assume that the presets are
broken and it won't hurt to break them further.

On Tue, Mar 29, 2016 at 2:41 PM, Austin S. Hemmelgarn <
notifications@github.com> wrote:

> I would vote yes, but that's just me.
>
> Ideally, I'd really like to see opinions on this from @hatstand
> <https://github.com/hatstand> and/or @ArnaudBienner
> <https://github.com/ArnaudBienner>.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/clementine-player/Clementine/issues/5286#issuecomment-203021388>
>



-- 
Rafael Giusti
Laboratório de Inteligência Computacional - LABIC
Universidade de São Paulo - USP

**Comment by Ferroin** [comment]
That really depends on what you mean by minimal.  Compared to changing the bands, yeah, it's essentially nothing, but it's still something that _will_ break user presets.

Arguments I've heard for continuing to use the current bands are:
1. It gives better control.
2. Winamp and Audacious use them.
3. It will break user presets.

Arguments I've seen for moving to the ISO bands:
1. It gives better control.
2. VLC, most media authoring software, and most pro audio hardware EQ's that provide a small number of bands use the ISO bands (and I believe that iTunes, WMP, Banshee, and Exaile use them also).
3. Our current presets are inherently broken as they were designed for the ISO bands, not what we currently use.

Obviously point one on both counts is almost entirely subjective.  Point two is overall in favor of switching as it is raising the concern of compatibility, and more things use the ISO bands.  Point three is the real issue here though.  Our current curves match those used on most ISO 266 EQs, but sound different because they aren't shifted for our bands.  Technically, due to aliasing and dithering, it's not possible to exactly match a curve intended for the ISO bands using our bands.  This in turn means that curves that have the same name in our software and (for example) VLC look the same, but sound noticeably different.

Most of my argument for it is based on the stronger compatibility and the fact that things are already broken (and IMO, the defaults we're shipping being broken is more significant than breaking user defined configurations, especially considering the fact that most of the people I've met either never use anything other than the defaults, or are constantly tweaking their configuration to fit the environment, and thus would end up adjusting to fit the new bands without much impact).
